drv/transceivers-server/src/udp.rs

```diff
@@ -270,53 +271,33 @@ impl ServerImpl {
-                        self.transceivers.clear_power_enable(mask).map_err(
-                            |_e| {
+                        self.transceivers
+                            .set_power_state(PowerState::A3, mask)
+                            .map_err(|_e| {
                                 Error::PowerModeFailed(
-                                    HwError::ClearPowerEnableFailed,
+                                    HwError::ControlPortWriteFailed,
                                 )
-                            },
-                        )?;
-                        self.transceivers.set_reset(mask).map_err(|_e| {
-                            Error::PowerModeFailed(HwError::SetResetFailed)
-                        })?;
-                        self.transceivers.set_lpmode(mask).map_err(|_e| {
-                            Error::PowerModeFailed(HwError::SetLpModeFailed)
-                        })?;
+                            })?;
                     }
                     PowerMode::Low => {
-                        // Power enabled, LpMode enabled
-                        self.transceivers.set_lpmode(mask).map_err(|_e| {
-                            Error::PowerModeFailed(HwError::SetLpModeFailed)
-                        })?;
-                        self.transceivers.set_power_enable(mask).map_err(
-                            |_e| {
+                        self.transceivers
+                            .set_power_state(PowerState::A2, mask)
+                            .map_err(|_e| {
                                 Error::PowerModeFailed(
-                                    HwError::SetPowerEnableFailed,
+                                    HwError::ControlPortWriteFailed,
                                 )
-                            },
-                        )?;
-                        self.transceivers.clear_reset(mask).map_err(|_e| {
-                            Error::PowerModeFailed(HwError::ClearResetFailed)
-                        })?;
+                            })?;
                     }
                     PowerMode::High => {
-                        // Power enabled, LpMode disabled
-                        self.transceivers.clear_lpmode(mask).map_err(|_e| {
-                            Error::PowerModeFailed(HwError::ClearLpModeFailed)
-                        })?;
-                        self.transceivers.set_power_enable(mask).map_err(
-                            |_e| {
+                        self.transceivers
+                            .set_power_state(PowerState::A0, mask)
+                            .map_err(|_e| {
                                 Error::PowerModeFailed(
-                                    HwError::SetPowerEnableFailed,
+                                    HwError::ControlPortWriteFailed,
                                 )
-                            },
-                        )?;
+                            })?;
```

Comment:
Should we wait / check here that it succeeds in going to A0?  That would match the behavior of the main sequencers.

Reply:
We could. My reasoning in not doing so here is that we would need a timeout policy or something on how long to wait until we give up on a `Fault`ing module before we return. Since we are sequencing up to 32 things in parallel here, it felt a bit different than our other sequencers. I tried to give the host more information via `Status` now to communicate power faults should they occur. Would it make sense to implement a `HostRequest::GetPowerMode` that would read the actual `STATUS_PORTx` and let the host do the comparison / fault handling?

Reply:
I'm on the fence, but am generally fine with keeping Hubris simple and punting this upstream to the host.